qt/python/mantidqt/mantidqt/widgets/helpwindow/helpwindowmodel.py

```diff
@@ -31,50 +50,110 @@ def interceptRequest(self, info: QWebEngineUrlRequestInfo):
+        """
+        Sets the internal state (_isLocal, _modeString, _baseWrl, _versionString)
+        based on the validity of localDocsBase and attempts to find a versioned URL.
+        """
+        if self._rawLocalDocsBase and os.path.isdir(self._rawLocalDocsBase):
+            self._isLocal = True
+            self._modeString = self.MODE_LOCAL
+            absLocalPath = os.path.abspath(self._rawLocalDocsBase)
+            self._baseUrl = QUrl.fromLocalFile(absLocalPath).toString()
+            self._versionString = None
+            logger.debug(f"Using {self._modeString} from {self._baseUrl}")
+        else:
+            if self._rawLocalDocsBase:
+                logger.warning(f"Local docs path '{self._rawLocalDocsBase}' is invalid or not found. Falling back to online docs.")
 
-    def is_local_docs_enabled(self):
+            self._isLocal = False
+            self._modeString = self.MODE_ONLINE
+
+            isLikelyRelease = self._rawLocalDocsBase is None
+            if isLikelyRelease:
+                self._versionString = getMantidVersionString()
+
+            if self._versionString:
+                baseOnline = self._rawOnlineBase
+                if baseOnline.endswith("/stable"):
+                    baseOnline = baseOnline[: -len("/stable")]
+
+                if self._versionString not in baseOnline:
+                    self._baseUrl = f"{baseOnline.rstrip('/')}/{self._versionString}"
+                    logger.debug(f"Using {self._modeString} (Version: {self._versionString}) from {self._baseUrl}")
+                else:
+                    self._baseUrl = self._rawOnlineBase
+                    logger.debug(f"Using {self._modeString} (Using provided base URL, possibly stable/latest): {self._baseUrl}")
+
+            else:
+                self._baseUrl = self._rawOnlineBase
+                logger.debug(f"Using {self._modeString} (Version: Unknown/Stable) from {self._baseUrl}")
+
+    def is_local_docs_mode(self):
         """
-        :return: True if local_docs_base is set and is a valid directory
+        :return: True if using local docs, False otherwise. Based on initial check.
         """
-        return bool(self.local_docs_base and os.path.isdir(self.local_docs_base))
+        return self._isLocal
 
-    def build_help_url(self, relative_url):
+    def get_mode_string(self):
         """
-        Returns a QUrl pointing to either local or online docs for the given relative URL.
+        :return: User-friendly string indicating the mode ("Local Docs" or "Online Docs").
         """
-        if not relative_url or not relative_url.endswith(".html"):
-            relative_url = "index.html"
+        return self._modeString
 
-        if self.is_local_docs_enabled():
-            full_path = os.path.join(self.local_docs_base, relative_url)
-            return QUrl.fromLocalFile(full_path)
-        else:
-            return QUrl(self.online_base + "/" + relative_url)
+    def get_base_url(self):
```

Comment:
If this is an internal method, maybe it would be better to fixup the base url in the constructor then change things that use it to directly use `self._baseUrl`